darts/models/forecasting/forecasting_model.py

```diff
@@ -1935,6 +2011,24 @@ def _get_default_model_params(cls):
             if p.name != "self"
         }
 
+    def _optimized_historical_forecasts(
+        self,
+        series: Sequence[TimeSeries],
+        past_covariates: Optional[Sequence[TimeSeries]] = None,
+        future_covariates: Optional[Sequence[TimeSeries]] = None,
+        num_samples: int = 1,
+        start: Optional[Union[pd.Timestamp, float, int]] = None,
+        forecast_horizon: int = 1,
+        stride: int = 1,
+        overlap_end: bool = False,
+        last_points_only: bool = True,
+        verbose: bool = False,
+        show_warnings: bool = True,
+    ) -> Union[
+        TimeSeries, List[TimeSeries], Sequence[TimeSeries], Sequence[List[TimeSeries]]
+    ]:
+        pass
```

Comment:
To prevent user from calling this method directly? Wouldn't be more efficient to just directly raise an error?

Aside from raising the error, what else would you expect this method to do? Because even if the sanity check pass, it's the sub-class `_optimized_historical_forecasts()` which is responsible for calling the appropriate optimized method.